src/core/pool_func.cpp

```diff
@@ -38,3 +42,19 @@
 		if (pool->type & pt) pool->CleanPool();
 	}
 }
+
+/* static */ void NORETURN PoolBase::Error(const char *format, ...)
+{
+	va_list ap;
+	char msg[256];
+
+	va_start(ap, format);
+	vseprintf(msg, lastof(msg), format, ap);
+	va_end(ap);
+
+	if (_switch_mode == SM_LOAD_GAME || _switch_mode == SM_LOAD_SCENARIO) {
+		SlErrorCorrupt(msg);
+	} else {
+		usererror("Failed loading savegame: %s", msg);
```

Comment:
this confuses me - you have a check that you're currently loading a game/scenario, but then this other message includes "loading savegame"

They can't both be correct?

Reply:
Er... good point o_O. At the point when I was checking we are loading a save that text had got lost from the usererror, but was put back later. Hmm!